File: 3159-find-occurrences-of-an-element-in-an-array/3159-find-occurrences-of-an-element-in-an-array.py
class Solution:
    def occurrencesOfElement(self, nums: List[int], queries: List[int], x: int) -> List[int]:
        pos = []
        for i in range(len(nums)):
            if nums[i] == x:
                pos.append(i)
        ans = []
        for i in queries:
                ans.append(-1 if i> len(pos) else pos[i-1])
        return ans

File: 3159-find-occurrences-of-an-element-in-an-array/3159-find-occurrences-of-an-element-in-an-array.cpp
class Solution {
public:
    vector<int> occurrencesOfElement(vector<int>& nums, vector<int>& queries, int x) {
        vector<int> pos,ans;
        for(int i=0;i<nums.size();i++){
            if(nums[i] == x) pos.push_back(i);
        }
        for(int i=0;i<queries.size();i++){
            if(pos.size() < queries[i]) ans.push_back(-1);
            else
            ans.push_back(pos[queries[i] - 1]);
        }
        return ans;
    }
};